### PR TITLE
add kaggle titanic example

### PR DESCRIPTION
**Pull request by yangarbiter**
built an example for titanic competition on kaggle

### CODE REVIEW

File: examples/titanic/run_model.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import numpy as np
+import pandas as pd
+from sklearn.ensemble import RandomForestClassifier
+
+def main():
+    data_csv_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
+    concat_feature_hdf_path = os.path.join(data_csv_dir, 'concat_feature.h5')
+    concat_f = h5py.File(concat_feature_hdf_path, 'r')
+    global_feature_hdf_path = os.path.join(data_csv_dir, 'data.h5')
+    global_f = h5py.File(global_feature_hdf_path, 'r')
```
Comment (ianlini) [inline examples/titanic/run_model.py]: unused line

File: examples/titanic/run_model.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import h5py
+import numpy as np
+import pandas as pd
+from sklearn.ensemble import RandomForestClassifier
+
+def main():
+    data_csv_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
+    concat_feature_hdf_path = os.path.join(data_csv_dir, 'concat_feature.h5')
+    concat_f = h5py.File(concat_feature_hdf_path, 'r')
+    global_feature_hdf_path = os.path.join(data_csv_dir, 'data.h5')
```
Comment (ianlini) [inline examples/titanic/run_model.py]: unsued line

File: examples/titanic/run_model.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    prediction_csv_path = 'prediction.csv'
+
+    is_valid = np.array(concat_f['valid_filter_list']['is_validation'])
+    is_test = np.array(concat_f['test_filter_list']['is_test'])
+    passenger_id = np.array(concat_f['passenger_id']['passenger_id'])
+    feature = np.array(concat_f['feature'])
+    label = np.array(concat_f['label']['label'])
+
+    concat_f.close()
+    global_f.close()
```
Comment (ianlini) [inline examples/titanic/run_model.py]: unused line

File: examples/titanic/run_model.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    train_filter = (np.bitwise_and(is_valid == 0, is_test == 0))
+    valid_filter = (np.bitwise_and(is_valid == 1, is_test == 0))
+    test_filter = (is_test == 1)
```
Comment (ianlini) [inline examples/titanic/run_model.py]: very bad, refer to the lifetime prediction example

File: examples/titanic/run_model.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    ##############
+    # validation #
+    ##############
+    clf = RandomForestClassifier()
+    clf.fit(feature[train_filter], label[train_filter])
+    print('validation score:',
+          clf.score(feature[valid_filter], label[valid_filter]))
```
Comment (ianlini) [inline examples/titanic/run_model.py]: what score?

File: examples/titanic/run_model.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    ##############
+    # prediction #
+    ##############
+    prediction = clf.predict(feature[test_filter])
+
+    df = pd.DataFrame(prediction, columns=['Survived'],
+                      index=passenger_id[test_filter])
+    df.index.name = 'PassengerId'
```
Comment (ianlini) [inline examples/titanic/run_model.py]: use rename

**Review by ianlini — CHANGES_REQUESTED** [review]

File: examples/titanic/run_model.py (hunk 5 above)
Comment (yangarbiter) [reply examples/titanic/run_model.py]: The score defined by  sklearn?  or I should use accuracy?

Comment (ianlini) [reply examples/titanic/run_model.py]: You should say what the score is

File: examples/titanic/README.rst [hunk 7]
```diff
@@ -0,0 +1,20 @@
+*******
+Titanic
+*******
```
Comment (ianlini) [inline examples/titanic/README.rst]: I use `=` for the first title, and only one line

**Review by ianlini — CHANGES_REQUESTED** [review]
It's better to split into 2 files

**Comment by ianlini** [comment]
Are you going to fix this? 
Switch to use the new API and command line tool.
